app/policies/person_policy.rb

```diff
@@ -85,7 +85,16 @@ def can_broker_modify?
   end
 
   def broker_agency_profile_matches?
-    associated_family.active_broker_agency_account.present? && associated_family.active_broker_agency_account.benefit_sponsors_broker_agency_profile_id == role.benefit_sponsors_broker_agency_profile_id
+    agency_id = role.benefit_sponsors_broker_agency_profile_id
+
+    family_active_broker = associated_family&.active_broker_agency_account
+    family_active_broker_matches = family_active_broker.present? && family_active_broker.benefit_sponsors_broker_agency_profile_id == agency_id
+
+    active_er = associated_family.primary_person&.active_employee_roles&.first
+    employer_active_broker = active_er&.employer_profile&.active_broker_agency_account
+    employer_active_broker_matches = employer_active_broker.present? && employer_active_broker.benefit_sponsors_broker_agency_profile_id == agency_id
+
+    family_active_broker_matches || employer_active_broker_matches
```

Comment:
Good job on using descriptive variable names. The use of safe navigation operator is effective. Noticed some code duplication when checking if `family_active_broker` and `employer_active_broker` match `agency_id`. Consider refactoring this into a private method to adhere to the DRY principle. Here's a suggestion:
```
private

def broker_matches?(broker, agency_id)
  broker.present? && broker.benefit_sponsors_broker_agency_profile_id == agency_id
end
```
And then use this method in broker_agency_profile_matches?:
```
def broker_agency_profile_matches?
  agency_id = role&.benefit_sponsors_broker_agency_profile_id

  family_active_broker = associated_family&.active_broker_agency_account
  active_er = associated_family.primary_person&.active_employee_roles&.first
  employer_active_broker = active_er&.employer_profile&.active_broker_agency_account

  broker_matches?(family_active_broker, agency_id) || broker_matches?(employer_active_broker, agency_id)
end
```